http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -168,6 +170,10 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
                     clusterRequest: ClusterRequest): Future[Cluster] = {
     val traceId = TraceId(randomUUID)
     for {
+    // Disallow users creating `highcpu` machine type with free credit projects
+      _ <- if(googleProject.value.startsWith("fccredits") && clusterRequest.machineConfig.exists(_.masterMachineType.exists(_.contains("highcpu"))))
```

Comment:
Geraldine has some concerns about this; some workshops require `highcpu` clusters for legitimate purposes. It might be better to put specific disallowed machine types in leo.conf, which we could change as needed.